lib/src/http/cdn/cdn_asset.dart

```diff
@@ -0,0 +1,110 @@
+import 'dart:typed_data';
+
+import 'package:nyxx/src/client.dart';
+import 'package:nyxx/src/http/cdn/cdn_request.dart';
+import 'package:nyxx/src/http/response.dart';
+import 'package:nyxx/src/http/route.dart';
+
+/// Available formats for CDN endpoints.
+enum CdnFormat {
+  /// The PNG format.
+  png._('png'),
+
+  /// The JPEG/JPG format.
+  jpeg._('jpeg'),
+
+  /// The webp format.
+  webp._('webp'),
+
+  /// The GIF format.
+  ///
+  /// This is only available to endpoints where [CdnAsset.isAnimated] is true.
+  gif._('gif'),
+
+  /// The Lottie format.
+  ///
+  /// This is only available to sticker endpoints where [Sticker.formatType] is [StickerFormatType.lottie].
+  lottie._('lottie');
+
+  /// The extension to use on the CDN endpoint for this format.
+  final String extension;
+
+  const CdnFormat._(this.extension);
+
+  @override
+  String toString() => 'CdnFormat($extension)';
+}
+
+/// {@template cdn_asset}
+/// An asset, most commonly an image, on Discord's CDN.
+/// {@endtemplate}
+class CdnAsset {
```

Comment:
can you add method that created just an url for an asset? That it could be sent that way via message?

Reply:
There already is - see the `url` getter below.

Reply:
im blind, all right the